core/src/main/java/tc/oc/pgm/tntrender/TNTRenderMatchModule.java

```diff
@@ -0,0 +1,101 @@
+package tc.oc.pgm.tntrender;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.bukkit.Effect;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.entity.TNTPrimed;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityExplodeEvent;
+import org.bukkit.event.entity.ExplosionPrimeEvent;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.nms.NMSHacks;
+
+public class TNTRenderMatchModule implements MatchModule, Listener {
+
+  private static final double MAX_DISTANCE = Math.pow(64d, 2);
+
+  private final Match match;
+  private final List<PrimedTnt> entities = new ArrayList<>();
```

Comment:
Move to constructor?

Reply:
Also, can we clear it on `disable`?

Reply:
I'll move it to constructor, i can not clear it on disable or it'd leave tnt hanging if match ended before it explodes for people far away, which reminds me i need to change the executor from running to LOADED. 
As the TNT finishes blowing up it will cleanup itself.